src/SignalR/clients/ts/signalr/src/IHubProtocol.ts

```diff
@@ -40,9 +36,7 @@ export type HubMessage =
     CompletionMessage |
     CancelInvocationMessage |
     PingMessage |
-    CloseMessage |
-    StreamCompleteMessage |
-    StreamDataMessage;
+    CloseMessage;
```

Comment:
Should InvocationMessage and StreamInvocationMessage also be combined?

Reply:
Couldn't that be determined based on the signature of the hub method being invoked? Or the presence of the stream/streamIds property in the InvocationMessage?

Reply:
Current design of the streamIds property is that it is only for streams from client to server.

And we can't rely on the signature of the Hub method to determine if the client intended to wait for a normal invocation response (via CompletionMessage) or a streamed response (via StreaItems + CompletionMessage). So we use two different invocation messages, StreamInvocationMessage and InvocationMessage